src/main/java/org/reaktivity/nukleus/http/internal/routable/stream/SourceInputStreamFactory.java

```diff
@@ -695,31 +700,39 @@ private void beforeWindowOrReset(
             }
         }
 
-        private void processInitialWindow(DirectBuffer buffer, int index, int length)
+        private void processWindowBeforeAlignment(DirectBuffer buffer, int index, int length)
         {
             windowRO.wrap(buffer, index, index + length);
             int update = windowRO.update();
             doSourceWindow(update);
+            availableTargetWindow += update;
+            processDeferredData();
+        }
 
-            // Process any deferred source data
-            int dataLength = Math.min(slotPosition, update);
+        private void processDeferredData()
+        {
+            int bytesToWrite = Math.min(slotPosition - slotOffset, availableTargetWindow);
             MutableDirectBuffer data = slab.buffer(slotIndex);
-            decode(data, slotOffset, dataLength);
-            if (dataLength < slotPosition)
-            {
-                slotOffset += dataLength;
-            }
-            else
+            decode(data, slotOffset, slotOffset + bytesToWrite);
+            availableTargetWindow -= bytesToWrite;
+
+            // Continue slabbing incoming data until target window updates have caught up
+            // with the initial window we gave to source
+            slotOffset += bytesToWrite;
+            int bytesLeft = slotPosition - slotOffset;
+            if (sourceUpdateDeferred >= 0 && bytesLeft == 0)
```

Comment:
I was having some difficulty following this logic at first, then realized that `slotPosition - slotOffset` is the size of bytes deferred in the slab.

Suggest the following change for lines `714`...
```
int bytesDeferred = slotPosition - slotOffset;
int bytesToWrite = Math.min(bytesDeferred, availableTargetWindow);
```
 and `721-723`...
```
bytesDeferred -= bytesToWrite;
if (sourceUpdateDeferred >= 0 && bytesDeferred == 0)
```

Also consider using the more familiar term `writableBytes` instead of `bytesToWrite`.